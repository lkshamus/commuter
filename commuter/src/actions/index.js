export const setCurrentWeather = (weather) => {
  return {
    type: 'SET_CURRENT_WEATHER',
    weather
  }
}

export const isLoading = (bool) => {
  return {
    type: 'IS_LOADING',
    isLoading: bool
  }
}

export const updateSearchWeather = (searchWeather) => {
  return {
    type: 'UPDATE_WEATHER_SEARCH',
    searchWeather
  }
}

export const setDirections = (directions) => {
  return {
    type: 'SET_DIRECTION',
    directions
  }
}

export const updateDirections = (searchedDirections) => {
  return {
    type: 'UPDATE_DIRECTIONS',
  searchedDirections
  }
}

export const setTransitDirections = (transitDirections) => {
  return {
    type: 'SET_TRANSIT_DIRECTIONS', 
    transitDirections
  }
}

export const setBikeDirections = (bikeDirections) => {
  return {
    type: 'SET_BIKE_DIRECTIONS',
    bikeDirections
  }
}

export const setWalkingDirections = (walkingDirections) => {
  return {
    type: 'SET_WALKING_DIRECTIONS', 
    walkingDirections
  }
}