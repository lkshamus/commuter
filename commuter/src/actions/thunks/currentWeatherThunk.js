import fetchCall from '../../utilities/fetchCall';
import * as cleaner from '../../utilities/cleaner';
import { setCurrentWeather } from '../index'

export const getCurrentWeather = () => {
  return async (dispatch) => {
    // dispatch(isLoading(true));
    try {
      const weather = await fetchCall(cleaner.searchWeatherByCity);
      console.log(weather)
      // if (weather === 'failed') {
      //   dispatch(setIsOk(true));
      // }
      // dispatch(isLoading(false));
      return dispatch(setCurrentWeather(weather));
    } catch(error) {
      console.log(error.message)
      // dispatch(setHasErrored(true));
    }
  }
}