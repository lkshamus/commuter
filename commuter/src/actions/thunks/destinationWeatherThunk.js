import fetchCall from '../../utilities/fetchCall';
import * as cleaner from '../../utilities/cleaner';
import { isLoading, updateSearchWeather, setHasErrored, setHasFailed } from '../index'

export const getDestinationWeather = (city) => {
  return async (dispatch) => {
    dispatch(isLoading(true));
    try {
      const weather = await fetchCall(cleaner.hourlyWeatherByCity(city));
      if (weather === 'failed') {
        dispatch(setHasFailed(true));
      }
      dispatch(isLoading(false));
      return dispatch(updateSearchWeather(weather));
    } catch(error) {
      console.log(error.message)
      dispatch(setHasErrored(true));
    }
  }
}
