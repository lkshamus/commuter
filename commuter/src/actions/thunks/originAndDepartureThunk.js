// import fetchCall from '../../utilities/fetchCall';
// import * as cleaner from '../../utilities/cleaner';
// import { isLoading, setDirections } from '../index'

// export const getCurrentDrivingDirections = (origin, departure) => {
//   return async (dispatch) => {
//     dispatch(isLoading(true));
//     try {
//       const directions = await fetchCall(cleaner.orignAndDepartureDriving(origin, departure));
//       console.log('Driving', directions)
//       // if (weather === 'failed') {
//       //   dispatch(setIsOk(true));
//       // }
//       dispatch(isLoading(false));
//       return dispatch(setDirections(directions));
//     } catch(error) {
//       console.log('this is the error block')
//       console.log(error.message)
//       // dispatch(setHasErrored(true));
//     }
//   }
// }
