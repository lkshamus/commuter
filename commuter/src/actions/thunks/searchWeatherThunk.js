import fetchCall from '../../utilities/fetchCall';
import * as cleaner from '../../utilities/cleaner';
import { isLoading, setCurrentWeather, updateSearchWeather } from '../index'

export const getCurrentWeather = (city) => {
  return async (dispatch) => {
    dispatch(isLoading(true));
    try {
      const weather = await fetchCall(cleaner.searchWeatherByCity(city));
      console.log('weather results:', weather)
      // if (weather === 'failed') {
      //   dispatch(setIsOk(true));
      // }
      dispatch(isLoading(false));
      return dispatch(setCurrentWeather(weather));
    } catch(error) {
      console.log(error.message)
      // dispatch(setHasErrored(true));
    }
  }
}
