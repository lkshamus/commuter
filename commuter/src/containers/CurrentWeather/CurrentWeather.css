.weather, .start-weather {
  display: flex;
  justify-content: start;
}

/*.start-weather {
  margin-bottom: 10px;
  margin-right: 20px;
}*/

.redo {
  position: relative;
  background-color: #c4d4e0;
  top: 350px;
  right: 600px;
  color: #666;
  font: 300 16px/1.7 'Open Sans', sans-serif;
  border: 0;
  outline: none;
  width: 90px;
  height: 50px;
  border-radius: 50px;
}

.redo:hover {
  outline: none;
  color: #ffffff;
  background-color: #193446;
}

.redo:disabled {
  background: #f1f1f1;
  color: #666;
}

.result {
  display: block;
}