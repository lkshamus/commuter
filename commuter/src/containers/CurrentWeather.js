import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { getCurrentWeather} from '../actions/thunks/currentWeatherThunk'
import { setCurrentWeather } from '../actions'
import Loading from '../components/Loading/index'
export class CurrentWeather extends Component {

 async componentDidMount() {
    await this.props.displayWeather(this.props.currentWeather)
  }

  render () {
    try{

       return (
      <div>
        <h1>weather in {this.props.currentWeather.name}</h1>
        <h1>weather in {Math.round((this.props.currentWeather.main.temp - 273.15) * 9 / 5 + 32)}Â°F</h1>
      </div>
    )

    } catch {
      return (
      <Loading />
      )
    }
  }
}

export const mapStateToProps = (state) => ({
  currentWeather: state.currentWeather,
  isLoading: state.isLoading
});

export const mapDispatchToProps = (dispatch) => ({
  displayWeather: (data) => dispatch(getCurrentWeather(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(CurrentWeather);