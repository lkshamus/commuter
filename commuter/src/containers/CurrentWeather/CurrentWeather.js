import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { NavLink } from "react-router-dom";

import { getCurrentWeather} from '../../actions/thunks/currentWeatherThunk'
import { setCurrentWeather } from '../../actions'
import Loading from '../../components/Loading/index'
import './CurrentWeather.css'

export class CurrentWeather extends Component {

 async componentDidMount() {
    await this.props.displayWeather(this.props.currentWeather)
  }

  render () {
    try{
       return (
      <div className='weather'>
        <h3> <span className='text'> Weather in </span> {this.props.currentWeather.name}</h3>
        <h3> <span className='text'> Weather in </span>{Math.round((this.props.currentWeather.main.temp - 273.15) * 9 / 5 + 32)}Â°F</h3>
        <h3> <span className='text'> Current Conditions: </span> {this.props.currentWeather.weather[0].description}</h3>
        <NavLink to='/'> <button>Redo Search</button></NavLink>
      </div>
    )
    } catch {
      return (
      <Loading />
      )
    }
  }
}

export const mapStateToProps = (state) => ({
  currentWeather: state.currentWeather,
  isLoading: state.isLoading
});

export const mapDispatchToProps = (dispatch) => ({
  displayWeather: (data) => dispatch(getCurrentWeather(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(CurrentWeather);