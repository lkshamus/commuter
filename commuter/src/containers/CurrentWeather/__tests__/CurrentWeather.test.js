/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { connect } from 'react-redux';

import { getDestinationWeather } from '../../../actions/thunks/destinationWeatherThunk'
import { getCurrentWeather } from '../../../actions/thunks/searchWeatherThunk'
import { CurrentWeather, mapDispatchToProps, mapStateToProps } from '../CurrentWeather';
import { setCurrentWeather } from '../../../actions'
import { updateSearchWeather } from '../../../actions'

describe('CurrentWeather', () => {
  it('should match snapshot', () => {
    const currentWeather = 
    {
      coord: {
        lon: 145.77,
        lat: -16.92
      },
        weather: 
        [{
          id: 802,
          main: "Clouds",
          description: "scattered clouds",
          icon: "03n"
        }],
          base: "stations",
          main: {
          temp: 300.15,
          pressure: 1007,
          humidity: 74,
          temp_min: 300.15,
          temp_max: 300.15
      },
    }

    const destinationWeather = 
    {
      cod: "200",
      message: 0.0036,
      cnt: 40,
      list: 
      [{
        dt: 1485799200,
        main: {
        temp: 261.45,
        temp_min: 259.086,
        temp_max: 261.45,
      },
        weather: 
        [{
          id: 800,
          main: "Clear",
          description: "clear sky",
          icon: "02n"
        }],
          clouds: {
          all: 8
        },
          wind: {
          speed: 4.77,
          deg: 232.505
        },
          snow: { },
          sys: {
          pod: "n"
        },
          dt_txt: "2017-01-30 18:00:00"
        },
      ]}
      
    const wrapper = shallow(<CurrentWeather destinationWeather={destinationWeather} currentWeather={currentWeather}/>)
    expect(wrapper).toMatchSnapshot();
  })
})

describe('mapDispatchToProps', () => {
  it('calls dispatch with a setCurrentWeather action when displayWeather is called', () => {

    const currentWeather =  {city: 'Mountain View', temp: 70}
      
    const mockDispatch = jest.fn()
    const actionToDispatch = setCurrentWeather(currentWeather)
      
    const mappedProps = mapDispatchToProps(mockDispatch)
    mappedProps.displayWeather(currentWeather)
    
    expect(mockDispatch).toHaveBeenCalled
  })
})


describe('mapStateToProps', () => {
  it('should map the state to props', () => {

    const currentWeather =  {city: 'Mountain View', temp: 70}

    const expected = {currentWeather}

    const mockState = {
      currentWeather: currentWeather
    }
    
    const mappedProps = mapStateToProps(mockState)
    expect (mappedProps).toEqual(expected)

  })
})