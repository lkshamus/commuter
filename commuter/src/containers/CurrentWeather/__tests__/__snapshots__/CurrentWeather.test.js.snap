// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentWeather should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CurrentWeather
    currentWeather={
      Object {
        "base": "stations",
        "coord": Object {
          "lat": -16.92,
          "lon": 145.77,
        },
        "main": Object {
          "humidity": 74,
          "pressure": 1007,
          "temp": 300.15,
          "temp_max": 300.15,
          "temp_min": 300.15,
        },
        "weather": Array [
          Object {
            "description": "scattered clouds",
            "icon": "03n",
            "id": 802,
            "main": "Clouds",
          },
        ],
      }
    }
    destinationWeather={
      Object {
        "cnt": 40,
        "cod": "200",
        "list": Array [
          Object {
            "clouds": Object {
              "all": 8,
            },
            "dt": 1485799200,
            "dt_txt": "2017-01-30 18:00:00",
            "main": Object {
              "temp": 261.45,
              "temp_max": 261.45,
              "temp_min": 259.086,
            },
            "snow": Object {},
            "sys": Object {
              "pod": "n",
            },
            "weather": Array [
              Object {
                "description": "clear sky",
                "icon": "02n",
                "id": 800,
                "main": "Clear",
              },
            ],
            "wind": Object {
              "deg": 232.505,
              "speed": 4.77,
            },
          },
        ],
        "message": 0.0036,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="weather-section"
      >
        <h2
          className="weather-header"
        >
          Weather
        </h2>
        <div
          className="start-weather"
        >
          <h3
            className="directions-weather"
          >
             
            <span
              className="text"
            >
               Origin weather: 
            </span>
            <span
              className="result"
            >
              81
              °F
            </span>
          </h3>
          <h3
            className="directions-weather"
          >
             
            <span
              className="text"
            >
               Current Conditions: 
            </span>
            <span
              className="result"
            >
              scattered clouds
              <img
                className="icon"
                src="http://openweathermap.org/img/w/03n.png"
              />
            </span>
          </h3>
        </div>
        <div
          className="weather"
        >
          <h3
            className="directions-weather"
          >
             
            <span
              className="text"
            >
               Destination weather: 
            </span>
            <span
              className="result"
            >
              11
              °F
            </span>
          </h3>
          <h3
            className="directions-weather"
          >
             
            <span
              className="text"
            >
               Current Conditions:  
            </span>
            <span
              className="result"
            >
              clear sky
              <img
                className="icon"
                src="http://openweathermap.org/img/w/02n.png"
              />
            </span>
          </h3>
        </div>
        <div
          className="new-search-btn"
        >
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/"
          >
             
            <button
              className="redo"
            >
              New Search
            </button>
          </NavLink>
        </div>
      </div>,
      "className": "weather-box",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="weather-header"
          >
            Weather
          </h2>,
          <div
            className="start-weather"
          >
            <h3
              className="directions-weather"
            >
               
              <span
                className="text"
              >
                 Origin weather: 
              </span>
              <span
                className="result"
              >
                81
                °F
              </span>
            </h3>
            <h3
              className="directions-weather"
            >
               
              <span
                className="text"
              >
                 Current Conditions: 
              </span>
              <span
                className="result"
              >
                scattered clouds
                <img
                  className="icon"
                  src="http://openweathermap.org/img/w/03n.png"
                />
              </span>
            </h3>
          </div>,
          <div
            className="weather"
          >
            <h3
              className="directions-weather"
            >
               
              <span
                className="text"
              >
                 Destination weather: 
              </span>
              <span
                className="result"
              >
                11
                °F
              </span>
            </h3>
            <h3
              className="directions-weather"
            >
               
              <span
                className="text"
              >
                 Current Conditions:  
              </span>
              <span
                className="result"
              >
                clear sky
                <img
                  className="icon"
                  src="http://openweathermap.org/img/w/02n.png"
                />
              </span>
            </h3>
          </div>,
          <div
            className="new-search-btn"
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/"
            >
               
              <button
                className="redo"
              >
                New Search
              </button>
            </NavLink>
          </div>,
        ],
        "className": "weather-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Weather",
            "className": "weather-header",
          },
          "ref": null,
          "rendered": "Weather",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="directions-weather"
              >
                 
                <span
                  className="text"
                >
                   Origin weather: 
                </span>
                <span
                  className="result"
                >
                  81
                  °F
                </span>
              </h3>,
              <h3
                className="directions-weather"
              >
                 
                <span
                  className="text"
                >
                   Current Conditions: 
                </span>
                <span
                  className="result"
                >
                  scattered clouds
                  <img
                    className="icon"
                    src="http://openweathermap.org/img/w/03n.png"
                  />
                </span>
              </h3>,
            ],
            "className": "start-weather",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  <span
                    className="text"
                  >
                     Origin weather: 
                  </span>,
                  <span
                    className="result"
                  >
                    81
                    °F
                  </span>,
                ],
                "className": "directions-weather",
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Origin weather: ",
                    "className": "text",
                  },
                  "ref": null,
                  "rendered": " Origin weather: ",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      81,
                      "°F",
                    ],
                    "className": "result",
                  },
                  "ref": null,
                  "rendered": Array [
                    81,
                    "°F",
                  ],
                  "type": "span",
                },
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  <span
                    className="text"
                  >
                     Current Conditions: 
                  </span>,
                  <span
                    className="result"
                  >
                    scattered clouds
                    <img
                      className="icon"
                      src="http://openweathermap.org/img/w/03n.png"
                    />
                  </span>,
                ],
                "className": "directions-weather",
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Current Conditions: ",
                    "className": "text",
                  },
                  "ref": null,
                  "rendered": " Current Conditions: ",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "scattered clouds",
                      <img
                        className="icon"
                        src="http://openweathermap.org/img/w/03n.png"
                      />,
                    ],
                    "className": "result",
                  },
                  "ref": null,
                  "rendered": Array [
                    "scattered clouds",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "icon",
                        "src": "http://openweathermap.org/img/w/03n.png",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                  ],
                  "type": "span",
                },
              ],
              "type": "h3",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="directions-weather"
              >
                 
                <span
                  className="text"
                >
                   Destination weather: 
                </span>
                <span
                  className="result"
                >
                  11
                  °F
                </span>
              </h3>,
              <h3
                className="directions-weather"
              >
                 
                <span
                  className="text"
                >
                   Current Conditions:  
                </span>
                <span
                  className="result"
                >
                  clear sky
                  <img
                    className="icon"
                    src="http://openweathermap.org/img/w/02n.png"
                  />
                </span>
              </h3>,
            ],
            "className": "weather",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  <span
                    className="text"
                  >
                     Destination weather: 
                  </span>,
                  <span
                    className="result"
                  >
                    11
                    °F
                  </span>,
                ],
                "className": "directions-weather",
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Destination weather: ",
                    "className": "text",
                  },
                  "ref": null,
                  "rendered": " Destination weather: ",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      11,
                      "°F",
                    ],
                    "className": "result",
                  },
                  "ref": null,
                  "rendered": Array [
                    11,
                    "°F",
                  ],
                  "type": "span",
                },
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  <span
                    className="text"
                  >
                     Current Conditions:  
                  </span>,
                  <span
                    className="result"
                  >
                    clear sky
                    <img
                      className="icon"
                      src="http://openweathermap.org/img/w/02n.png"
                    />
                  </span>,
                ],
                "className": "directions-weather",
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Current Conditions:  ",
                    "className": "text",
                  },
                  "ref": null,
                  "rendered": " Current Conditions:  ",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "clear sky",
                      <img
                        className="icon"
                        src="http://openweathermap.org/img/w/02n.png"
                      />,
                    ],
                    "className": "result",
                  },
                  "ref": null,
                  "rendered": Array [
                    "clear sky",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "icon",
                        "src": "http://openweathermap.org/img/w/02n.png",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                  ],
                  "type": "span",
                },
              ],
              "type": "h3",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              to="/"
            >
               
              <button
                className="redo"
              >
                New Search
              </button>
            </NavLink>,
            "className": "new-search-btn",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": Array [
                " ",
                <button
                  className="redo"
                >
                  New Search
                </button>,
              ],
              "to": "/",
            },
            "ref": null,
            "rendered": Array [
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "New Search",
                  "className": "redo",
                },
                "ref": null,
                "rendered": "New Search",
                "type": "button",
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="weather-section"
        >
          <h2
            className="weather-header"
          >
            Weather
          </h2>
          <div
            className="start-weather"
          >
            <h3
              className="directions-weather"
            >
               
              <span
                className="text"
              >
                 Origin weather: 
              </span>
              <span
                className="result"
              >
                81
                °F
              </span>
            </h3>
            <h3
              className="directions-weather"
            >
               
              <span
                className="text"
              >
                 Current Conditions: 
              </span>
              <span
                className="result"
              >
                scattered clouds
                <img
                  className="icon"
                  src="http://openweathermap.org/img/w/03n.png"
                />
              </span>
            </h3>
          </div>
          <div
            className="weather"
          >
            <h3
              className="directions-weather"
            >
               
              <span
                className="text"
              >
                 Destination weather: 
              </span>
              <span
                className="result"
              >
                11
                °F
              </span>
            </h3>
            <h3
              className="directions-weather"
            >
               
              <span
                className="text"
              >
                 Current Conditions:  
              </span>
              <span
                className="result"
              >
                clear sky
                <img
                  className="icon"
                  src="http://openweathermap.org/img/w/02n.png"
                />
              </span>
            </h3>
          </div>
          <div
            className="new-search-btn"
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/"
            >
               
              <button
                className="redo"
              >
                New Search
              </button>
            </NavLink>
          </div>
        </div>,
        "className": "weather-box",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="weather-header"
            >
              Weather
            </h2>,
            <div
              className="start-weather"
            >
              <h3
                className="directions-weather"
              >
                 
                <span
                  className="text"
                >
                   Origin weather: 
                </span>
                <span
                  className="result"
                >
                  81
                  °F
                </span>
              </h3>
              <h3
                className="directions-weather"
              >
                 
                <span
                  className="text"
                >
                   Current Conditions: 
                </span>
                <span
                  className="result"
                >
                  scattered clouds
                  <img
                    className="icon"
                    src="http://openweathermap.org/img/w/03n.png"
                  />
                </span>
              </h3>
            </div>,
            <div
              className="weather"
            >
              <h3
                className="directions-weather"
              >
                 
                <span
                  className="text"
                >
                   Destination weather: 
                </span>
                <span
                  className="result"
                >
                  11
                  °F
                </span>
              </h3>
              <h3
                className="directions-weather"
              >
                 
                <span
                  className="text"
                >
                   Current Conditions:  
                </span>
                <span
                  className="result"
                >
                  clear sky
                  <img
                    className="icon"
                    src="http://openweathermap.org/img/w/02n.png"
                  />
                </span>
              </h3>
            </div>,
            <div
              className="new-search-btn"
            >
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/"
              >
                 
                <button
                  className="redo"
                >
                  New Search
                </button>
              </NavLink>
            </div>,
          ],
          "className": "weather-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Weather",
              "className": "weather-header",
            },
            "ref": null,
            "rendered": "Weather",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3
                  className="directions-weather"
                >
                   
                  <span
                    className="text"
                  >
                     Origin weather: 
                  </span>
                  <span
                    className="result"
                  >
                    81
                    °F
                  </span>
                </h3>,
                <h3
                  className="directions-weather"
                >
                   
                  <span
                    className="text"
                  >
                     Current Conditions: 
                  </span>
                  <span
                    className="result"
                  >
                    scattered clouds
                    <img
                      className="icon"
                      src="http://openweathermap.org/img/w/03n.png"
                    />
                  </span>
                </h3>,
              ],
              "className": "start-weather",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    <span
                      className="text"
                    >
                       Origin weather: 
                    </span>,
                    <span
                      className="result"
                    >
                      81
                      °F
                    </span>,
                  ],
                  "className": "directions-weather",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": " Origin weather: ",
                      "className": "text",
                    },
                    "ref": null,
                    "rendered": " Origin weather: ",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        81,
                        "°F",
                      ],
                      "className": "result",
                    },
                    "ref": null,
                    "rendered": Array [
                      81,
                      "°F",
                    ],
                    "type": "span",
                  },
                ],
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    <span
                      className="text"
                    >
                       Current Conditions: 
                    </span>,
                    <span
                      className="result"
                    >
                      scattered clouds
                      <img
                        className="icon"
                        src="http://openweathermap.org/img/w/03n.png"
                      />
                    </span>,
                  ],
                  "className": "directions-weather",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": " Current Conditions: ",
                      "className": "text",
                    },
                    "ref": null,
                    "rendered": " Current Conditions: ",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "scattered clouds",
                        <img
                          className="icon"
                          src="http://openweathermap.org/img/w/03n.png"
                        />,
                      ],
                      "className": "result",
                    },
                    "ref": null,
                    "rendered": Array [
                      "scattered clouds",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "icon",
                          "src": "http://openweathermap.org/img/w/03n.png",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "img",
                      },
                    ],
                    "type": "span",
                  },
                ],
                "type": "h3",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3
                  className="directions-weather"
                >
                   
                  <span
                    className="text"
                  >
                     Destination weather: 
                  </span>
                  <span
                    className="result"
                  >
                    11
                    °F
                  </span>
                </h3>,
                <h3
                  className="directions-weather"
                >
                   
                  <span
                    className="text"
                  >
                     Current Conditions:  
                  </span>
                  <span
                    className="result"
                  >
                    clear sky
                    <img
                      className="icon"
                      src="http://openweathermap.org/img/w/02n.png"
                    />
                  </span>
                </h3>,
              ],
              "className": "weather",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    <span
                      className="text"
                    >
                       Destination weather: 
                    </span>,
                    <span
                      className="result"
                    >
                      11
                      °F
                    </span>,
                  ],
                  "className": "directions-weather",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": " Destination weather: ",
                      "className": "text",
                    },
                    "ref": null,
                    "rendered": " Destination weather: ",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        11,
                        "°F",
                      ],
                      "className": "result",
                    },
                    "ref": null,
                    "rendered": Array [
                      11,
                      "°F",
                    ],
                    "type": "span",
                  },
                ],
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    <span
                      className="text"
                    >
                       Current Conditions:  
                    </span>,
                    <span
                      className="result"
                    >
                      clear sky
                      <img
                        className="icon"
                        src="http://openweathermap.org/img/w/02n.png"
                      />
                    </span>,
                  ],
                  "className": "directions-weather",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": " Current Conditions:  ",
                      "className": "text",
                    },
                    "ref": null,
                    "rendered": " Current Conditions:  ",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "clear sky",
                        <img
                          className="icon"
                          src="http://openweathermap.org/img/w/02n.png"
                        />,
                      ],
                      "className": "result",
                    },
                    "ref": null,
                    "rendered": Array [
                      "clear sky",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "icon",
                          "src": "http://openweathermap.org/img/w/02n.png",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "img",
                      },
                    ],
                    "type": "span",
                  },
                ],
                "type": "h3",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                to="/"
              >
                 
                <button
                  className="redo"
                >
                  New Search
                </button>
              </NavLink>,
              "className": "new-search-btn",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": Array [
                  " ",
                  <button
                    className="redo"
                  >
                    New Search
                  </button>,
                ],
                "to": "/",
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "New Search",
                    "className": "redo",
                  },
                  "ref": null,
                  "rendered": "New Search",
                  "type": "button",
                },
              ],
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
